Src/ILGPU/Context.cs

```diff
@@ -384,6 +385,77 @@ public DeviceCollection<TDevice> GetDevices<TDevice>()
                 : new DeviceCollection<TDevice>(new List<Device>());
         }
 
+        /// <summary>
+        /// Attempts to return the most optimal single device.
+        /// </summary>
+        /// <param name="PreferCPU">Always returns CPU device 0.</param>
+        /// <returns>Selected device.</returns>
+        public Device GetBestDevice(bool PreferCPU = false)
+        {
+            if(PreferCPU)
+            {
+                return deviceMapping.TryGetValue(AcceleratorType.CPU, out var devices)
+                    ? devices.First()
+                    : throw new NotSupportedException(
+                            RuntimeErrorMessages.NotSupportedTargetAccelerator);
+            }
+
+            var sorted = Devices.Sort((d1, d2) => d1.MemorySize.CompareTo(d2.MemorySize))
```

Comment:
nit: Coding style, break on `.Sort`

Reply:
SUGGESTION: It might fit on the line if replaced by `.OrderByDescending(d => d.MemorySize)`

Reply:
Changed to OrderByDecending I don't know what I used sort